.github/workflows/main.yaml

```
@@ -0,0 +1,164 @@
+name: main
+
+on:
+  workflow_call:
+    inputs:
+      build-artifact-name:
+        required: true
+        type: string
+      build-artifact-path:
+        required: true
+        type: string
+      self-build:
+        required: false
+        type: number
+      pyodide-version:
+        required: false
+        type: string
+        default: "0.21.0"
+      runner:
+        required: false
+        type: string
+        default: "selenium"
+      browser:
+        required: false
+        type: string
+        default: "chrome"
+      browser-version:
+        required: false
+        type: string
+        default: "latest"
+      os:
+        required: false
+        type: string
+        default: "ubuntu-latest"
+      driver-version:
+        required: false
+        type: string
+        default: "latest"
+      refresh:
+        required: false
+        type: string
+        default: "0"
+permissions:
+  contents: read
+
+jobs:
+  test:
+    name: test ${{ inputs.browser }} ( ${{ inputs.runner }},${{ inputs.os }},pyodide:${{ inputs.pyodide-version}})
+    runs-on: ${{ inputs.os }}
+    env:
+      DISPLAY: :99
+    steps:
+      - uses: actions/checkout@v2
+      - name: Cache Pyodide
+        uses: actions/cache@v3
+        id: cache-pyodide
+        with:
+          path: pyodide-dist
+          key: pyodide-${{ inputs.pyodide-version }}-${{ hashFiles('.github/**/*.yaml') }}
+      - name: Download Pyodide
+        shell: bash -l {0}
+        if: steps.cache-pyodide.outputs.cache-hit != 'true'
+        run: |
+          wget -q https://github.com/pyodide/pyodide/releases/download/${{ inputs.pyodide-version }}/pyodide-build-${{ inputs.pyodide-version }}.tar.bz2
+          tar xjf pyodide-build-${{ inputs.pyodide-version }}.tar.bz2
+          mv pyodide pyodide-dist/
+      - name: Download build artifacts from calling package
+        if: inputs.self-build!=1
+        uses: actions/download-artifact@v3
+        with:
+          name: ${{ inputs.build-artifact-name }}
+          path: ${{ inputs.build-artifact-path }}
+      - uses: actions/setup-python@v3
+        with:
+          python-version: 3.10.2
+      - name: Install node
+        uses: actions/setup-node@v3
+        if: ${{ contains(inputs.browser, 'node') || inputs.runner == 'playwright' }}
+        with:
+          node-version: ${{ inputs.driver-version }}
+
+      - name: Cache Playwright browsers
+        uses: actions/cache@v3
+        if: ${{ inputs.runner == 'playwright' }}
+        with:
+          path: .cache/ms-playwright
+          key: ${{ runner.os }}-playwright-${{ inputs.browser-version }}
+
+      - name: Install playwright
+        shell: bash -l {0}
+        if: ${{ inputs.runner == 'playwright' }}
+        run: |
+          python3 -m pip install playwright==${{ inputs.browser-version }}
```

```suggestion
          python3 -m pip install playwright
```

I think we should not use browser version variable for playwright version. For now let's use latest version and add variable for this later.